test/e2e/e2e_test.go

```diff
@@ -184,3 +321,46 @@ func registerBroker(namespace string) error {
 
 	return err
 }
+
+// setupInstance will create an instance for each test case in their own unique namespace
+func setupServiceInstance(t *testing.T, namespace string) (*v1beta1.ServiceInstance, error) {
+	// cluster resource definition interface for service instance
+	crd := &v1beta1.ServiceInstance{}
+	testutil.ReadInYAMLFileAndConvert("../../samples/kubernetes/instance.yaml", &crd)
+
+	// Provision instance from the CRD using the service catalog
+	provisionedCRD, err := svcatClient.ServicecatalogV1beta1().ServiceInstances(namespace).Create(crd)
+	if !assert.NoError(t, err) {
+		return nil, err
+	}
+
+	// Naming convention used in kubernetes
+	serviceinstance := &v1beta1.ServiceInstance{}
+	// Wait up to 25 minutes for the service instance to have been provisioned.
+	err = testutil.Poll(25, func() (bool, error) {
```

Comment:
am I right that `Poll` returns error if the specified time has passed and the callback hasn't returned true?